handler/src/main/java/io/netty/handler/ssl/OpenSsl.java

```diff
@@ -57,7 +60,22 @@
         // If in the classpath, try to load the native library and initialize netty-tcnative.
         if (cause == null) {
             try {
-                NativeLibraryLoader.load("netty-tcnative", SSL.class.getClassLoader());
+                String os = normalizeOs(SystemPropertyUtil.get("os.name", ""));
+                String arch = normalizeArch(SystemPropertyUtil.get("os.arch", ""));
+                String libName = "netty-tcnative-" + os + '-' + arch;
+                try {
+                    // First, try loading the platform-specific library. Platform-specific
+                    // libraries will be available if using a tcnative uber jar.
+                    NativeLibraryLoader.load(libName, SSL.class.getClassLoader());
```

Comment:
@nmittler @normanmaurer any thoughts on how we might approach this for openssl and the fedora library? yet another try catch? bit of a special case since the artifact only exists for openssl, not boringssl.